components/dialog/dialog-mixin.js

```diff
@@ -177,10 +177,33 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		}
 	}
 
+	_findAutofocusElement(node) {
+		if (this._useNative) {
+			// Do not override native autofocus attribute implementation
+			return null;
+		}
+
+		if (!node.querySelector('slot')) {
+			// We are in a confirm dialog, autofocus attribute will never be set
+			return null;
+		}
```

Comment:
I don't really like that this is here, because it limits future use cases. I have left it in because it currently suits our needs and there is no use case of `d2l-dialog` that wouldn't pass this guard. The alternative is to add extra logic for composing slots conditionally.